lib/service/servicecfg/database.go

```diff
@@ -99,6 +100,17 @@ func (d *Database) CheckAndSetDefaults() error {
 		}
 	}
 
+	// If AWS account ID is missing, but assume role ARN is given,
+	// try to parse the role arn and set the account id to match.
+	if d.AWS.AccountID == "" && d.AWS.AssumeRoleARN != "" {
+		parsed, err := awsutils.ParseRoleARN(d.AWS.AssumeRoleARN)
+		if err != nil {
+			return trace.BadParameter("database %q invalid AWS assume_role_arn: %v",
+				d.Name, err)
+		}
+		d.AWS.AccountID = parsed.AccountID
+	}
+
```

Comment:
nit: Not a problem started by this change but ideally we should do this in api.Database's `CheckAndSetDefaults` so that databases created by tctl and webapi can benefit from it. (maybe the same for the TLS and AD `CheckAndSetDefaults` above, if we can move them to api if possible)

@smallinsky @GavinFrazar do you think if worth copying the AWS arn implementation into our api lib?
https://github.com/aws/aws-sdk-go/blob/v1.44.224/aws/arn/arn.go
It's not that much code. And if we have our own implementation, we can even enhance it with more functions we need.

Reply:
👍  Yeah I put it here to avoid pulling AWS sdk into our api. I have the same check in `ValidateDatabase`, but it just errors for empty AccountID because we don't set defaults in that function. Would definitely be a better UX for tctl and webapi if we can move these all into the api `CheckAndSetDefaults`.

Reply:
> do you think if worth copying the AWS arn implementation into our api lib?
https://github.com/aws/aws-sdk-go/blob/v1.44.224/aws/arn/arn.go

Yea we can do that but why not just embed the aws.ARN into own custom type and add additional logic.

Reply:
> Yea we can do that but why not just embed the aws.ARN into own custom type and add additional logic ?

I think we hesitate whether to bring the whole AWS SDK dependency to api's go.mod. 

Another alternative is to get `aws-sdk-go-v2` without other services.

Reply:
I think I want to just pull in `aws-sdk-go-v2` since the stripped binary size change is tiny and v2 doesn't download the whole sdk when fetching dependencies, but I should move this code in another PR to get feedback from other people first. It enables only a small UX improvement for `tctl` and web api. I can move the TLS and AD stuff too in that PR.